### PR TITLE
CT-8552 Added method to get abstract file children count by type

### PR DESCRIPTION
**Pull request by eugene7646**

### CODE REVIEW

File: bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java [hunk 1]
```diff
@@ -5495,6 +5495,75 @@ int getContentChildrenCount(Content content) throws TskCoreException {
...
+	/**
+	 * Count of the content object's children of specified types. The types are
+	 * instances of TskData.TSK_FS_NAME_TYPE_ENUM types and the matching is
+	 * performed against the tsk_files.dir_type column. Some usage examples are
+	 * to get a count of all subdirectories, which requires searching for all
+	 * children of types TskData.TSK_FS_NAME_TYPE_ENUM.DIR OR
+	 * TskData.TSK_FS_NAME_TYPE_ENUM.VIRT_DIR.
+	 *
+	 * Note: this is generally more efficient then preloading all children and
+	 * counting, and facilities lazy loading.
```
Comment (APriestman) [inline bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java:5507]: I realize you cut and pasted this from above, but I think it should be "facilitates"

File: bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java [hunk 2]
```diff
@@ -5495,6 +5495,75 @@ int getContentChildrenCount(Content content) throws TskCoreException {
...
+			String sql = "SELECT COUNT(*) AS count "
+					+ "FROM tsk_objects "
+					+ "INNER JOIN tsk_files ON tsk_objects.obj_id = tsk_files.obj_id "
+					+ "WHERE (tsk_objects.par_obj_id = ? AND tsk_files.dir_type IN (" + inClause.toString() + "))";
+
+			PreparedStatement statement = connection.getConnection().prepareStatement(sql);
+			statement.clearParameters();
+			statement.setLong(1, content.getId());
+
+            for (int i = 0; i < types.size(); i++) {
```
Comment (APriestman) [inline bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java:5548]: Inconsistent indentation here

File: bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java (hunk 1 above)
Comment (eugene7646) [reply bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java]: Fixed

File: bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java (hunk 2 above)
Comment (eugene7646) [reply bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java]: It only looks this way in github for some reason. The indentation is correct is Netbeans.

**Review by APriestman — APPROVED** [review]

File: bindings/java/src/org/sleuthkit/datamodel/AbstractFile.java [hunk 3]
```diff
@@ -908,6 +908,26 @@ public boolean isDir() {
...
+	 *
+	 * @param types List of TskData.TSK_FS_NAME_TYPE_ENUM types.
+	 *
+	 * @return Total count of children of the specified types.
+	 *
+	 * @throws org.sleuthkit.datamodel.TskCoreException exception thrown if a
+	 *                                                  critical error occurs
+	 *                                                  within tsk core
+	 */
+	public int countChildrenOfType(List<TSK_FS_NAME_TYPE_ENUM> types) throws TskCoreException {
```
Comment (gdicristofaro) [inline bindings/java/src/org/sleuthkit/datamodel/AbstractFile.java:928]: Does this need to be marked as `@Beta` by any chance?

Comment (eugene7646) [reply bindings/java/src/org/sleuthkit/datamodel/AbstractFile.java:928]: No idea when we apply this marking. @APriestman ?

Comment (APriestman) [reply bindings/java/src/org/sleuthkit/datamodel/AbstractFile.java:928]: I've been putting it on most of my new stuff just so we're not committing to all the new APIs :)

Comment (eugene7646) [reply bindings/java/src/org/sleuthkit/datamodel/AbstractFile.java:930]: How non-committal of us! ;) 

Done.

**Review by gdicristofaro — APPROVED** [review]

**Pull request merged**